debezium-core/src/main/java/io/debezium/jdbc/JdbcValueConverters.java

```diff
@@ -325,23 +339,11 @@ public ValueConverter converter(Column column, Field fieldDefn) {
 
             // Date and time values
             case Types.DATE:
-                if (adaptiveTimePrecisionMode || adaptiveTimeMicrosecondsPrecisionMode) {
-                    return (data) -> convertDateToEpochDays(column, fieldDefn, data);
-                }
-                return (data) -> convertDateToEpochDaysAsDate(column, fieldDefn, data);
+                return (data) -> convertDate(column, fieldDefn, data);
             case Types.TIME:
                 return (data) -> convertTime(column, fieldDefn, data);
             case Types.TIMESTAMP:
-                if (adaptiveTimePrecisionMode || adaptiveTimeMicrosecondsPrecisionMode) {
-                    if (getTimePrecision(column) <= 3) {
-                        return data -> convertTimestampToEpochMillis(column, fieldDefn, data);
-                    }
-                    if (getTimePrecision(column) <= 6) {
-                        return data -> convertTimestampToEpochMicros(column, fieldDefn, data);
-                    }
-                    return (data) -> convertTimestampToEpochNanos(column, fieldDefn, data);
-                }
-                return (data) -> convertTimestampToEpochMillisAsDate(column, fieldDefn, data);
+                return (data) -> convertTimestamp(column, fieldDefn, data);
```

Comment:
moved the code to `convertTimestamp` method so it could be reused buy sub-classes.